node-graph/graph-craft/src/proto.rs

```diff
@@ -19,6 +19,7 @@ pub type DynFuture<'n, T> = Pin<Box<dyn core::future::Future<Output = T> + 'n>>;
 pub type LocalFuture<'n, T> = Pin<Box<dyn core::future::Future<Output = T> + 'n>>;
 pub type Any<'n> = Box<dyn DynAny<'n> + 'n>;
 pub type FutureAny<'n> = DynFuture<'n, Any<'n>>;
+// TODO: is this safe? This is assumed to be send+sync.
```

Comment:
In the past I required `TypeErasedNodes` to be `Send+Sync` but this does not work when executing futures on the web so we basically would need a feature flag conditionally enabling the Send Sync bound for the type erasure. 
I just never bothered to actually implement that hence the current state. I still use arcs even though we only use a single thread because 1. I was too lazy to change all occurrences, 2. Because this is just compiled to an Rc anyway because we can only have one thread, 3. This indicates that we will want to support multi-threading in the future